e2e_test.go

```diff
@@ -0,0 +1,313 @@
+package e2etest
+
+import (
+	"context"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+
+	"github.com/openconfig/clio/collector"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/pmetric/pmetricotlp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	metricTimestamp = time.Date(2020, 2, 11, 20, 26, 12, 321, time.UTC)
+)
+
+// newTs advances the metricTimestamp and returns a new timestamp.
+func newTs() pcommon.Timestamp {
+	metricTimestamp = metricTimestamp.Add(time.Minute)
+	return pcommon.NewTimestampFromTime(metricTimestamp)
+}
+
+// configProviderSettings is a convenience function to create ConfigProviderSettings that use the
+// local config.yaml.
+func configProviderSettings() otelcol.ConfigProviderSettings {
+	return otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs: []string{filepath.Join("config", "config.yaml")},
+			ProviderFactories: []confmap.ProviderFactory{
+				fileprovider.NewFactory(),
+			},
+		},
+	}
+}
+
+// subscribeRequestForTarget is a convenience function to create a target-specific SubscribeRequest.
+func subscribeRequestForTarget(t *testing.T, target string) *gpb.SubscribeRequest {
+	return &gpb.SubscribeRequest{
+		Request: &gpb.SubscribeRequest_Subscribe{
+			Subscribe: &gpb.SubscriptionList{
+				Prefix: &gpb.Path{
+					Target: target,
+					Elem: []*gpb.PathElem{
+						{
+							Name: target,
+						},
+					},
+				},
+				Mode: gpb.SubscriptionList_STREAM,
+				Subscription: []*gpb.Subscription{
+					&gpb.Subscription{
+						Mode:              gpb.SubscriptionMode_ON_CHANGE,
+						Path:              &gpb.Path{},
+						SuppressRedundant: false,
+					},
+				},
+			},
+		},
+	}
+}
+
+// startCollectorPipeline starts and returns the collector pipeline (plus the WaitGroup it runs in).
+func startCollectorPipeline(t *testing.T, ctx context.Context) (*sync.WaitGroup, *otelcol.Collector) {
+	t.Helper()
+	t.Log("Starting collector pipeline")
+
+	// Create collector.
+	set := otelcol.CollectorSettings{
+		BuildInfo:              component.NewDefaultBuildInfo(),
+		Factories:              collector.Components,
+		ConfigProviderSettings: configProviderSettings(),
+	}
+	col, err := otelcol.NewCollector(set)
+	require.NoError(t, err)
+
+	// Start collector.
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		require.NoError(t, col.Run(ctx))
```

Comment:
Generally we tend to avoid assertion frameworks - could we do the same here?

Reply:
done.